Workflow failure: Keycloak CI - Store IT (aurora-postgres)

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

Recently during the weekend this stage started to fail with following message, occurred in 3 runs leading into the same result on each run.
```
The error appears to be in '/home/runner/work/keycloak/keycloak/.github/scripts/ansible/roles/mvn_ec2_runner/tasks/run.yml': line 15, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Run mvn command on the remote hosts
  ^ here
Error: Process completed with exit code 4.
``` 

job1: https://github.com/keycloak/keycloak/actions/runs/7937743626/job/21675720361
job2: https://github.com/keycloak/keycloak/actions/runs/7945178083/job/21691732542
job3: https://github.com/keycloak/keycloak/actions/runs/7952710409/job/21707852026

### Version

latest

### Regression

- [ ] The issue is a regressions

### Expected behavior

n/a

### Actual behavior

n/a

### How to Reproduce?

n/a

### Anything else?

_No response_

--- a/.github/scripts/ansible/roles/mvn_ec2_runner/tasks/run.yml
+++ b/.github/scripts/ansible/roles/mvn_ec2_runner/tasks/run.yml
@@ -17,7 +17,8 @@
     set -o pipefail
     cd {{ kc_home }}
     ./mvnw {{ mvn_params }}
-  executable: /usr/bin/bash
+  args:
+    executable: /usr/bin/bash
   # Tests can run for hours. To prevent the test from failing when the SSH connection breaks, use asynchronous polling.
   async: 86400
   poll: 10

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -372,7 +372,7 @@ jobs:
           ./aws_ec2.sh create ${REGION}
           ./keycloak_ec2_installer.sh ${REGION} /tmp/keycloak.zip
           ./mvn_ec2_runner.sh ${REGION} "clean install -B -DskipTests -Pdistribution"
-          ./mvn_ec2_runner.sh ${REGION} "clean install -B -DskipTests -pl testsuite/integration-arquillian/servers/auth-server/quarkus -Pauth-server-quarkus -Dmaven.build.cache.enabled=true"
+          ./mvn_ec2_runner.sh ${REGION} "clean install -B -DskipTests -pl testsuite/integration-arquillian/servers/auth-server/quarkus -Pauth-server-quarkus -Pdb-aurora-postgres -Dmaven.build.cache.enabled=true"
           ./mvn_ec2_runner.sh ${REGION} "test -B ${{ env.SUREFIRE_RETRY }} -Pauth-server-quarkus -Pdb-${{ matrix.db }} $PROPS -Dtest=$TESTS -pl testsuite/integration-arquillian/tests/base 2>&1 | misc/log/trimmer.sh"
 
           # Copy returned surefire-report directories to workspace root to ensure they're discovered

--- a/testsuite/integration-arquillian/pom.xml
+++ b/testsuite/integration-arquillian/pom.xml
@@ -505,19 +505,19 @@
         <profile>
             <id>db-aurora-postgres</id>
             <properties>
-                <keycloak.storage.connections.vendor>postgres</keycloak.storage.connections.vendor>
-                <keycloak.connectionsJpa.driver>org.postgresql.Driver</keycloak.connectionsJpa.driver>
+                <keycloak.storage.connections.vendor>aurora-postgres</keycloak.storage.connections.vendor>
+                <keycloak.connectionsJpa.driver>software.amazon.jdbc.Driver</keycloak.connectionsJpa.driver>
                 <keycloak.connectionsJpa.database>keycloak</keycloak.connectionsJpa.database>
                 <keycloak.connectionsJpa.user>keycloak</keycloak.connectionsJpa.user>
                 <keycloak.connectionsJpa.password>secret99</keycloak.connectionsJpa.password>
                 <keycloak.connectionsJpa.jdbcParameters/>
-                <keycloak.connectionsJpa.url>jdbc:postgresql://${auth.server.db.host}/${keycloak.connectionsJpa.database}${keycloak.connectionsJpa.jdbcParameters}</keycloak.connectionsJpa.url>
+                <keycloak.connectionsJpa.url>jdbc:aws-wrapper:postgresql://${auth.server.db.host}/${keycloak.connectionsJpa.database}${keycloak.connectionsJpa.jdbcParameters}</keycloak.connectionsJpa.url>
                 <!-- JDBC properties point to "default" JDBC driver for the particular DB -->
                 <!-- For EAP testing, it is recommended to override those with system properties pointing to GAV of more appropriate JDBC driver -->
                 <!-- for the particular EAP version -->
-                <jdbc.mvn.groupId>org.postgresql</jdbc.mvn.groupId>
-                <jdbc.mvn.artifactId>postgresql</jdbc.mvn.artifactId>
-                <jdbc.mvn.version>${postgresql-jdbc.version}</jdbc.mvn.version>
+                <jdbc.mvn.groupId>software.amazon.jdbc</jdbc.mvn.groupId>
+                <jdbc.mvn.artifactId>aws-advanced-jdbc-wrapper</jdbc.mvn.artifactId>
+                <jdbc.mvn.version>${aws-jdbc-wrapper.version}</jdbc.mvn.version>
             </properties>
         </profile>
         <profile>

--- a/testsuite/integration-arquillian/servers/auth-server/quarkus/pom.xml
+++ b/testsuite/integration-arquillian/servers/auth-server/quarkus/pom.xml
@@ -387,5 +387,36 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <id>db-aurora-postgres</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>includeProprietaryDependencies</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>copy</goal>
+                                </goals>
+                                <configuration>
+                                    <artifactItems>
+                                        <artifactItem>
+                                            <groupId>software.amazon.jdbc</groupId>
+                                            <artifactId>aws-advanced-jdbc-wrapper</artifactId>
+                                            <version>${aws-jdbc-wrapper.version}</version>
+                                            <type>jar</type>
+                                            <outputDirectory>${auth.server.home}/providers</outputDirectory>
+                                        </artifactItem>
+                                    </artifactItems>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
     </profiles>
 </project>

--- a/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/model/StoreProvider.java
+++ b/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/model/StoreProvider.java
@@ -31,26 +31,30 @@ public enum StoreProvider {
     JPA("jpa") {
         @Override
         public void addStoreOptions(List<String> commands) {
-            getDbVendor().ifPresent(vendor -> commands.add("--db=" + vendor));
-            commands.add("--db-username=" + System.getProperty("keycloak.connectionsJpa.user"));
-            commands.add("--db-password=" + System.getProperty("keycloak.connectionsJpa.password"));
-            if ("mssql".equals(getDbVendor().orElse(null))){
-                commands.add("--transaction-xa-enabled=false");
-            }
-            commands.add("--db-url=" + System.getProperty("keycloak.connectionsJpa.url"));
+            addOptions("--", commands);
         }
 
         @Override
         public List<String> getStoreOptionsToKeycloakConfImport() {
             List<String> options = new ArrayList<>();
-            getDbVendor().ifPresent(vendor -> options.add("db=" + vendor));
-            options.add("db-url=" + System.getProperty("keycloak.connectionsJpa.url"));
-            options.add("db-username=" + System.getProperty("keycloak.connectionsJpa.user"));
-            options.add("db-password=" + System.getProperty("keycloak.connectionsJpa.password"));
-            if ("mssql".equals(getDbVendor().orElse(null))){
-                options.add("transaction-xa-enabled=false");
-            }
+            addOptions("", options);
             return options;
+        }
+
+        private void addOptions(String prefix, List<String> options) {
+            getDbVendor().ifPresent(vendor -> {
+                if ("mssql".equals(vendor)) {
+                    options.add(prefix + "transaction-xa-enabled=false");
+                } else if ("aurora-postgres".equals(vendor)){
+                    vendor = "postgres";
+                    options.add(prefix + "db-driver=" + System.getProperty("keycloak.connectionsJpa.driver"));
+                    options.add(prefix + "transaction-xa-enabled=false");
+                }
+                options.add(prefix + "db=" + vendor);
+            });
+            options.add(prefix + "db-username=" + System.getProperty("keycloak.connectionsJpa.user"));
+            options.add(prefix + "db-password=" + System.getProperty("keycloak.connectionsJpa.password"));
+            options.add(prefix + "db-url=" + System.getProperty("keycloak.connectionsJpa.url"));
         }
     },
     DEFAULT("default") {

